typed-racket-lib/typed-racket/static-contracts/instantiate.rkt

```diff
@@ -211,16 +216,37 @@
                   #`(define #,id #,ctc)))
         ctc))
 
+(module predefined-contracts racket/base
+  ;; this table maps (cons Type? typed-side?) -> (cons identifier? kind?)
+  ;; where the identifier can be used instead of generating a contract for the type
+  (define predef-contracts (make-hash))
+  (define-syntax-rule (define/contract-decl i ctc sc)
+    (begin (hash-set! predef-contracts sc #'i)
+           (define i ctc)))
+  (provide predef-contracts define/contract-decl))
+
+(require (submod "." predefined-contracts)
+         (for-template (submod "." predefined-contracts)))
+
 ;; Determine whether the given contract syntax should be inlined or not.
 (define (should-inline-contract? stx)
-  (or
-   ;; no need to generate an extra def for things that are already identifiers
-   (identifier? stx)
-   ;; ->* are handled specially by the contract system
-   (let ([sexp (syntax-e stx)])
-     (and (pair? sexp)
-          (or (free-identifier=? (car sexp) #'->)
-              (free-identifier=? (car sexp) #'->*))))))
+   (syntax-case stx (quote)
+     ;; no need to generate an extra def for things that are already identifiers
+     [i (identifier? #'i) #t]
+     [(quote i) (or (identifier? #'i)
+                    (boolean? (syntax-e #'i))
+                    (number? (syntax-e #'i)))
+      #t]
+      [_ #f]))
+
+(define (function-contract? stx)
+  (syntax-case stx ()
+    [(arr . _)
+     (and (identifier? #'arr)
```

Comment:
this is really low importance, but I've been trying to avoid using `arr` because people so easily confuse "arrow", "arity", etc and conversations about them can be really confusing if taken literally when they are used incorrectly. maybe it doesn't matter here.